lib/widgets/home.dart

```diff
@@ -2,96 +2,533 @@ import 'package:flutter/material.dart';
-            const SizedBox(height: 16),
-            ElevatedButton(
-              onPressed: () => Navigator.push(context,
-                RecentDmConversationsPage.buildRoute(context: context)),
-              child: Text(zulipLocalizations.recentDmConversationsPageTitle)),
-            if (testStreamId != null) ...[
-              const SizedBox(height: 16),
-              ElevatedButton(
-                onPressed: () => Navigator.push(context,
-                  MessageListPage.buildRoute(context: context,
-                    narrow: ChannelNarrow(testStreamId!))),
-                child: const Text("#test here")), // scaffolding hack, see above
-            ],
+      appBar: ZulipAppBar(titleSpacing: 16,
+        title: Text(_currentTabTitle)),
+      body: Stack(
+        children: [
+          for (final (tab, body) in pageBodies)
+            // TODO(a11y): Decide if we find it helpful to use something like
+            //   [SemanticsProperties.namesRoute] to structure this UI better
+            //   for screen-reader software.
+            Offstage(offstage: tab != _tab.value, child: body),
+        ]),
+      bottomNavigationBar: DecoratedBox(
+        decoration: BoxDecoration(
+          border: Border(top: BorderSide(color: designVariables.borderBar, width: 0)),
+          color: designVariables.bgBotBar),
+        child: SafeArea(
+          // [Center] is necessary to limit the width of the row of icons,
+          // because it allows the child to be smaller than itself.
+          child: Center(
+            // This shrinks [Center] to be as tall as its child, instead of
+            // growing as tall as [Scaffold] allows, filling the entire screen.
+            heightFactor: 1,
+            child: ConstrainedBox(
+              // TODO(design): determine a suitable max width for bottom nav bar
+              constraints: const BoxConstraints(maxWidth: 600).tighten(height: 48),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceAround,
+                crossAxisAlignment: CrossAxisAlignment.stretch,
+                children: [
+                  for (final navigationButton in navigationButtons)
+                    Expanded(child: navigationButton),
+                ]))))));
+  }
+}
+
+class _LoadingPlaceholderPage extends StatelessWidget {
+  const _LoadingPlaceholderPage();
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Scaffold(
+      body: Column(
+        crossAxisAlignment: CrossAxisAlignment.end,
+        children: [
+          const Expanded(child: LoadingPlaceholder()),
+          TextButton(
+            onPressed: () {
+              Navigator.of(context).push(
+                MaterialWidgetRoute(page: const ChooseAccountPage()));
+            },
+            child: Text(zulipLocalizations.chooseAccountPageTitle)),
+        ]));
+  }
+}
+
+class _NavigationButton extends StatelessWidget {
```

Comment:
bump https://github.com/zulip/zulip-flutter/pull/1076#discussion_r1868535479

Reply:
I see. Missed the `Bar` part. Fixed that and updated comment/string references to this name to match.